src/main/java/seedu/address/model/person/Task.java

```diff
@@ -0,0 +1,139 @@
+package seedu.address.model.person;
+
+import java.util.Date;
+
+import seedu.address.model.person.exceptions.InvalidTimeException;
+
+/**
+ * Represents a piece of work for a staff.
+ */
+public class Task {
+    private String taskName;
+    private Time startTime;
+    private Time endTime;
+    private Date date;
+
+    /**
+     * Constructor of Task given date, startTime, endTime and taskName.
+     * @param date Date of the task.
+     * @param startTime StartTime of the task.
+     * @param endTime EndTime of the task.
+     * @param taskName Name of the task.
+     */
+    public Task(Date date, Time startTime, Time endTime, String taskName) {
+        this.date = date;
+        this.startTime = startTime;
+        this.endTime = endTime;
+        this.taskName = taskName;
+    }
+
+    /**
+     * Constructor of Task given date, startTime, and endTime.
+     * @param date Date of the task.
+     * @param startTime StartTime of the task.
+     * @param endTime EndTime of the task.
+     */
+    public Task(Date date, Time startTime, Time endTime) {
+        this.date = date;
+        this.startTime = startTime;
+        this.endTime = endTime;
+    }
+
+    /**
+     * Constructor of Task given date, name and time represented by hour and min.
+     * @param date Date of task.
+     * @param startHour StartHour of task.
+     * @param startMin StartMin of task.
+     * @param endHour EndHour of task.
+     * @param endMin EndMin of task.
+     * @param taskName Name of task.
+     * @throws InvalidTimeException Throws if the given hour and min are invalid.
+     */
+    public Task(Date date, int startHour, int startMin, int endHour, int endMin, String taskName)
+            throws InvalidTimeException {
+        this.date = date;
+        this.startTime = new Time(startHour, startMin);
+        this.endTime = new Time(endHour, endMin);
+        this.taskName = taskName;
+    }
+
+    /**
+     * Constructor of Task given date, and time represented by hour and min.
+     * @param date Date of task.
+     * @param startHour StartHour of task.
+     * @param startMin StartMin of task.
+     * @param endHour EndHour of task.
+     * @param endMin EndMin of task.
+     * @throws InvalidTimeException Throws if the given hour and min are invalid.
+     */
+    public Task(Date date, int startHour, int startMin, int endHour, int endMin) throws InvalidTimeException {
```

Comment:
When will this constructor (without task name) be used, and when will the constructor with taskname be used?

Reply:
Maybe when the staff doesn't want to specify the task name or he or she wants to? Do you think I need to add more comments or just delete one?